src/GitVersion.Core.Tests/Configuration/ConfigurationProviderTests.CanWriteOutEffectiveConfiguration.approved.txt

```diff
@@ -1,14 +1,27 @@
 assembly-versioning-scheme: MajorMinorPatch
 assembly-file-versioning-scheme: MajorMinorPatch
-mode: ContinuousDelivery
 tag-prefix: '[vV]'
-continuous-delivery-fallback-tag: ci
 major-version-bump-message: '\+semver:\s?(breaking|major)'
 minor-version-bump-message: '\+semver:\s?(feature|minor)'
 patch-version-bump-message: '\+semver:\s?(fix|patch)'
 no-bump-message: '\+semver:\s?(none|skip)'
 tag-pre-release-weight: 60000
-commit-message-incrementing: Enabled
+fallback-branch:
+  mode: ContinuousDelivery
+  tag: '{BranchName}'
+  increment: Inherit
+  prevent-increment-of-merged-branch-version: false
+  track-merge-target: false
+  commit-message-incrementing: Enabled
+  regex: ''
+  tracks-release-branches: false
+  is-release-branch: false
+  is-mainline: false
+unknown-branch:
+  mode: ContinuousDelivery
+  tag: '{BranchName}'
+  increment: Inherit
+  regex: ''
```

Comment:
I'm not sure if I understood you correct.  Do you mean the position or the name of this property?

Reply:
> Why isn't `unknown-branch` just an `unknown` property beneath the `branches` property?

What happened if the user wants to configure a branch with the name unknown? ;)

Reply:
The property name beneath `branches` is not the name of the actual branch in Git. The name of the actual branch in Git is identified in the child `regex` property. The property names beneath `branches` are virtual and identify the **branch category** which GitVersion has pre-existing knowledge about in order to provide out of the box versioning for widely used workflows such as Git Flow.

Reply:
Okay maybe we should rename it to something else. Please notice that I haven't introduce the branch name placeholder (see https://github.com/GitTools/GitVersion/commit/71d8fe5a8e8199cf36d7e523e45982b2dbd1f1d8#diff-ad2fc424e99560699ab35cf2da3f7b20b939f1ad927a2d5dd8519828f5a09161).

Are you agree with the introduction of the unknown branch section?

Reply:
Yes, having an `unknown` branch category makes sense and is something that is being discussed in https://github.com/GitTools/GitVersion/issues/2870#issuecomment-1315378556 as well.

Reply:
Of course you can add a configuration to the array and match any branches with the regex .* But if you doing this the order of the branch configuration array is important. Conceptionally you need to introduce a order by property. Otherwise it is hard or even impossible to override the branch configuration in GitVersion.yaml.

Exactly that is reason why I have introduced the unknown branch configuration.

Reply:
> Exactly that is reason why I have introduced the unknown branch configuration section to specify the configuration for branches which are not known.

Yes, exactly. That's why I write:

> Yes, having an `unknown` branch category makes sense.

:)

>The reason why I'm not sure if it is a good idea to move the unknown branch section in the branch configurations array is, that we have magic string and again special business logic which is not obviously to the user. 

Which "magic string" are you referring to?

>Introducing a order by property would be better but for our use case over engineered.

I would not agree to add an `order` property. We already have the idea of an `unknown` branch in the codebase today, it's just not clearly defined or named. Once we give it a name, it becomes much clearer both in the codebase and in the documentation what it is, how it works, etc.

Reply:
It makes no sense to define a regular expression like .* without a order by property. You are not able to add new branches in the GitVersion.yaml file because the .* configuration matches first. Or would you like to introduce a magic string like unknown and say: Please if a unknown branch name existing in the array then move this configuration to the last position. That's not transparent to the user. Sorry I don't see no concept to be able to configure branches who are unknown.

Reply:
`regex: .*` makes sense for the `unknown` branch category, because it will by definition be evaluated last by GitVersion. Even if `unknown` is not present in the configuration, GitVersion should have a concept of an `unkown` branch which has `regex: .*` by default. The `unknown` branch configuration is only needed to be present in `GitVersion.yml` if the user needs it to deviate from the defaults (as with any other branch configuraiton).